tests/test_annubes_env.py

```diff
@@ -100,7 +106,91 @@ def test_fix_time_types(time, expected_type):
         ), f"Expected {t} time to be a multiple of dt ({env.dt}), but got {env._duration['fixation']}"
 
 
-def test_observation_space(default_env, custom_env):
+@pytest.mark.parametrize("catch_prob", [0.0, 0.3, 0.7, 1.0])
+def test_catch_prob(catch_prob: float) -> None:
+    """Test if the catch trial probability is working as expected.
+
+    This test checks:
+    1. If the observed probability of catch trials matches the specified probability
+    2. If the probability works correctly for various values including edge cases (0.0 and 1.0)
+    """
+    n_trials = 1000
+    env = AnnubesEnv(catch_prob=catch_prob, random_seed=RND_SEED)
+    catch_count = 0
+
+    for _ in range(n_trials):
+        env.reset()
+        trial_info = env.trial
+        if trial_info["catch"]:
+            catch_count += 1
+
+    observed_prob = catch_count / n_trials
+    expected_prob = catch_prob
+
+    # Check if the observed probability is close to the expected probability
+    assert np.isclose(
+        observed_prob,
+        expected_prob,
+        atol=0.05,
+    ), f"Expected catch probability {expected_prob}, but got {observed_prob}"
+
+
+@pytest.mark.parametrize(
+    ("session", "catch_prob", "max_sequential"),
+    [
+        ({"v": 0.5, "a": 0.5}, 0.5, 3),
+        ({"v": 0.3, "a": 0.7}, 0.1, 4),
+        ({"v": 0.3, "a": 0.3, "av": 0.3}, 0.7, 4),
+    ],
+)
+def test_annubes_env_max_sequential(session: dict, catch_prob: float, max_sequential: int) -> None:
+    """Test the maximum sequential trial constraint in the AnnubesEnv.
+
+    The test performs the following checks:
+    1. Ensures no sequence of non-catch trials longer than `max_sequential`
+       of the same type occurs.
+    2. Verifies that all specified trial types in the session occur.
+    3. Checks if the distribution of trial types is roughly balanced
+       according to the specified probabilities.
+    4. Verifies that the frequency of catch trials matches the specified
+       `catch_prob`.
+    """
+    env = AnnubesEnv(session=session, catch_prob=catch_prob, max_sequential=max_sequential, random_seed=RND_SEED)
+
+    trial_types: list[str | None] = []
+    for _ in range(N_TRIALS):
+        env.new_trial()
+        trial_types.append(env.trial["stim_type"])
+
+    # Check for sequences longer than max_sequential, excluding None (catch trials)
+    for i in range(len(trial_types) - max_sequential):
+        sequence = [t for t in trial_types[i : i + max_sequential + 1] if t is not None]
+        if len(sequence) > max_sequential:
+            assert len(set(sequence)) > 1, f"Found a sequence longer than {max_sequential} at index {i}"
```

Comment:
This looks right to me (apart from the fact that i do think we should test for catches as well, see comment above). 
However, I remember in the previous repo we also had a similar looking test that ultimately turned out not to be accurate. Have you looked back at that one to see whether the mistake we made there was avoided here? You could also copy the test I wrote in the other repo to be sure that it works as expected.

Reply:
I double checked this one and it should work as expected, but in case you see a bug there let me know :)